controllers/storageautoscaler/util.go

```diff
@@ -0,0 +1,109 @@
+package storageautoscaler
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/go-logr/logr"
+	prometheusapi "github.com/prometheus/client_golang/api"
+	prometheusv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	prometheusconfig "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/platform"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func getPrometheusURL(operatorNamespace string) (string, error) {
+	prometheusURL := "https://prometheus-k8s.openshift-monitoring.svc.cluster.local:9091"
+	if isROSAHCP, err := platform.IsPlatformROSAHCP(); err != nil {
+		return "", err
+	} else if isROSAHCP {
+		prometheusURL = fmt.Sprintf("https://prometheus.%s.svc.cluster.local:9339", operatorNamespace)
+	}
+
+	return prometheusURL, nil
+}
+
+func getScrpaer(ctx context.Context, operatorNamespace string, log logr.Logger) (prometheusv1.API, error) {
+	prometheusURL, err := getPrometheusURL(operatorNamespace)
+	if err != nil {
+		log.Error(err, "Failed to determine if ROSA HCP cluster")
+		return nil, err
+	}
+
+	caCertPath := "/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt"
+	tokenPath := "/var/run/secrets/kubernetes.io/serviceaccount/token"
+
+	// create auth roundtripper
+	tlsConfig, err := prometheusconfig.NewTLSConfig(&prometheusconfig.TLSConfig{
+		CAFile: caCertPath,
+	})
+	if err != nil {
+		log.Error(err, "failed to create tls config")
+		return nil, err
+	}
+	settings := prometheusconfig.TLSRoundTripperSettings{}
+	newRT := func(cfg *tls.Config) (http.RoundTripper, error) {
+		return &http.Transport{TLSClientConfig: cfg}, nil
+	}
+	tlsRoundTripper, err := prometheusconfig.NewTLSRoundTripperWithContext(ctx, tlsConfig, settings, newRT)
+	if err != nil {
+		log.Error(err, "failed to create tls round tripper")
+		return nil, err
+	}
+	roundTripper := prometheusconfig.NewAuthorizationCredentialsRoundTripper("Bearer", prometheusconfig.NewFileSecret(tokenPath), tlsRoundTripper)
+	// create prometheus client
+	prometheusClient, err := prometheusapi.NewClient(prometheusapi.Config{
+		Address:      prometheusURL,
+		RoundTripper: roundTripper,
+	})
+	if err != nil {
+		log.Error(err, "failed to create prometheus client")
+		return nil, err
+	}
+	// scrape the metrics
+	scraper := prometheusv1.NewAPI(prometheusClient)
+
+	return scraper, nil
```

Comment:
How about some debug logs so we can see starting/completion of metrics scraping?

Reply:
I do have logs in `scraper.go` from where this function is called, actually there is no debug loggin in ocs operator so everything has to be in INFO